### PR TITLE
Refine unused warning for shorthand field patterns

### PR DESCRIPTION
**Pull request by luc-blaeser**
Refine the unused warning for shorthand field patterns. Until now, it suggested to prefix the identifier by '_' which however changes the referred field name and thus the type. Therefore, distinguish cases of unused shorthand field patterns from other declarations where the identifier can be renamed.

### Example
```
let { firstName; secondName } = person;
Debug.print(firstName);
// unused `secondName` identifier
```

### Warnings

Old misleading warning:
```
warning [M0194], unused identifier secondName (delete or rename to wildcard `_` or `_secondName`)
```

New improved warning:
```
warning [M0198], unused field secondName in object pattern (delete or rewrite as `secondName = _`)
```

### Suggestions

The refined warning (cf. extended description) suggests the following fixes:
1. Remove the unused field pattern (assign to super-type):
  
  ```
  let { firstName } = person;
  ```

2. Bind to wildcard:
  ```
  let { firstName; secondName = _ } = person;
  ```

3. Bind to `_`-prefixed identifier:
  ```
  let { firstName; secondName = _secondName } = person;
  ```

### CODE REVIEW

File: src/lang_utils/error_codes/M0198.md [hunk 1]
```diff
@@ -0,0 +1,54 @@
+# M0198
+
+This warning means that you specified a field identifier in an object pattern without referencing this identifier later, a good indicator of dead code.
+
+Dubious code example:
+
+```motoko
+import Debug "mo:base/Debug";
```
Comment (luc-blaeser) [inline src/lang_utils/error_codes/M0198.md:8]: Question: I am not sure whether this code is shown in the web-based Motoko interpreter and whether it supports base library imports.

Comment (crusso) [reply src/lang_utils/error_codes/M0198.md:8]: I think VS code will display it, but not sure about the interpreter. I wouldn't worry about it now though.

**Review by crusso — COMMENTED** [review]
Looks nice! Mind if I refactor a little to reduce the diff?

**Review by crusso — APPROVED** [review]
Actually, go ahead and merge, I can easily refactor in a separate PR.

A short changelog entry would be good.

**Comment by crusso** [comment]
Great PR description!

**Comment by luc-blaeser** [comment]
Thanks a lot, @crusso, for reviewing and refactoring!

**Pull request merged**